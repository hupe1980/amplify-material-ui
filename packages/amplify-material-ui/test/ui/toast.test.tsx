import * as React from 'react';
import { render } from '@testing-library/react';
import { Toast } from '../../src';
import { withTheme } from '../helpers/withTheme';

describe('loading', () => {
  it('should be rendered correctly (error)', () => {
    const { asFragment } = render(withTheme(<Toast content="Error" variant="error" open />));
    expect(asFragment()).toMatchSnapshot();
  });

  it('should be rendered correctly (info)', () => {
    const { asFragment } = render(withTheme(<Toast content="Info" variant="info" open />));
    expect(asFragment()).toMatchSnapshot();
  });

  it('should be rendered correctly (warning)', () => {
    const { asFragment } = render(withTheme(<Toast content="Warning" variant="warning" open />));
    expect(asFragment()).toMatchSnapshot();
  });

  it('should be rendered correctly (success)', () => {
    const { asFragment } = render(withTheme(<Toast content="Success" variant="success" open />));
    expect(asFragment()).toMatchSnapshot();
  });
});
