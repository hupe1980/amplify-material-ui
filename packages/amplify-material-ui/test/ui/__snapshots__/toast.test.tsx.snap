// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loading should be rendered correctly (error) 1`] = `
<DocumentFragment>
  <div
    class="MuiSnackbar-root MuiSnackbar-anchorOriginTopCenter css-zzms1-MuiSnackbar-root"
    role="presentation"
  >
    <div
      aria-describedby="client-snackbar"
      class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation6 MuiSnackbarContent-root makeStyles-error-2 css-1eqdgzv-MuiPaper-root-MuiSnackbarContent-root"
      direction="down"
      role="alert"
      style="opacity: 1; transform: scale(1, 1); transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,transform 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    >
      <div
        class="MuiSnackbarContent-message css-1exqwzz-MuiSnackbarContent-message"
      >
        <span
          class="makeStyles-message-7"
          id="client-snackbar"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium makeStyles-icon-5 makeStyles-iconVariant-6 css-i4bv87-MuiSvgIcon-root"
            data-testid="ErrorIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-2h2v2zm0-4h-2V7h2v6z"
            />
          </svg>
          Error
        </span>
      </div>
      <div
        class="MuiSnackbarContent-action css-1kr9x0n-MuiSnackbarContent-action"
      >
        <button
          aria-label="Close"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeLarge css-1pz7awu-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium makeStyles-icon-5 css-i4bv87-MuiSvgIcon-root"
            data-testid="CloseIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`loading should be rendered correctly (info) 1`] = `
<DocumentFragment>
  <div
    class="MuiSnackbar-root MuiSnackbar-anchorOriginTopCenter css-zzms1-MuiSnackbar-root"
    role="presentation"
  >
    <div
      aria-describedby="client-snackbar"
      class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation6 MuiSnackbarContent-root makeStyles-info-10 css-1eqdgzv-MuiPaper-root-MuiSnackbarContent-root"
      direction="down"
      role="alert"
      style="opacity: 1; transform: scale(1, 1); transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,transform 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    >
      <div
        class="MuiSnackbarContent-message css-1exqwzz-MuiSnackbarContent-message"
      >
        <span
          class="makeStyles-message-14"
          id="client-snackbar"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium makeStyles-icon-12 makeStyles-iconVariant-13 css-i4bv87-MuiSvgIcon-root"
            data-testid="InfoIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-6h2v6zm0-8h-2V7h2v2z"
            />
          </svg>
          Info
        </span>
      </div>
      <div
        class="MuiSnackbarContent-action css-1kr9x0n-MuiSnackbarContent-action"
      >
        <button
          aria-label="Close"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeLarge css-1pz7awu-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium makeStyles-icon-12 css-i4bv87-MuiSvgIcon-root"
            data-testid="CloseIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`loading should be rendered correctly (success) 1`] = `
<DocumentFragment>
  <div
    class="MuiSnackbar-root MuiSnackbar-anchorOriginTopCenter css-zzms1-MuiSnackbar-root"
    role="presentation"
  >
    <div
      aria-describedby="client-snackbar"
      class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation6 MuiSnackbarContent-root makeStyles-success-22 css-1eqdgzv-MuiPaper-root-MuiSnackbarContent-root"
      direction="down"
      role="alert"
      style="opacity: 1; transform: scale(1, 1); transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,transform 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    >
      <div
        class="MuiSnackbarContent-message css-1exqwzz-MuiSnackbarContent-message"
      >
        <span
          class="makeStyles-message-28"
          id="client-snackbar"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium makeStyles-icon-26 makeStyles-iconVariant-27 css-i4bv87-MuiSvgIcon-root"
            data-testid="CheckCircleIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm-2 15-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
            />
          </svg>
          Success
        </span>
      </div>
      <div
        class="MuiSnackbarContent-action css-1kr9x0n-MuiSnackbarContent-action"
      >
        <button
          aria-label="Close"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeLarge css-1pz7awu-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium makeStyles-icon-26 css-i4bv87-MuiSvgIcon-root"
            data-testid="CloseIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`loading should be rendered correctly (warning) 1`] = `
<DocumentFragment>
  <div
    class="MuiSnackbar-root MuiSnackbar-anchorOriginTopCenter css-zzms1-MuiSnackbar-root"
    role="presentation"
  >
    <div
      aria-describedby="client-snackbar"
      class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation6 MuiSnackbarContent-root makeStyles-warning-18 css-1eqdgzv-MuiPaper-root-MuiSnackbarContent-root"
      direction="down"
      role="alert"
      style="opacity: 1; transform: scale(1, 1); transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,transform 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    >
      <div
        class="MuiSnackbarContent-message css-1exqwzz-MuiSnackbarContent-message"
      >
        <span
          class="makeStyles-message-21"
          id="client-snackbar"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium makeStyles-icon-19 makeStyles-iconVariant-20 css-i4bv87-MuiSvgIcon-root"
            data-testid="WarningIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M1 21h22L12 2 1 21zm12-3h-2v-2h2v2zm0-4h-2v-4h2v4z"
            />
          </svg>
          Warning
        </span>
      </div>
      <div
        class="MuiSnackbarContent-action css-1kr9x0n-MuiSnackbarContent-action"
      >
        <button
          aria-label="Close"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeLarge css-1pz7awu-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium makeStyles-icon-19 css-i4bv87-MuiSvgIcon-root"
            data-testid="CloseIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
